util/winutil/mksyscall.go

```diff
@@ -6,5 +6,11 @@ package winutil
 //go:generate go run golang.org/x/sys/windows/mkwinsyscall -output zsyscall_windows.go mksyscall.go
 //go:generate go run golang.org/x/tools/cmd/goimports -w zsyscall_windows.go
 
+//sys getApplicationRestartSettings(process windows.Handle, commandLine *uint16, commandLineLen *uint32, flags *uint32) (ret wingoes.HRESULT) = kernel32.GetApplicationRestartSettings
```

Comment:
We'd have to make changes to the code generator. Part of the issue is that `HRESULT`s also support success codes, so if we were to naïvely go straight from `HRESULT` to `error`, we would break the idiom that successful function calls have `nil` errors.